### PR TITLE
Remove object snapshots, only program snapshots

### PR DESCRIPTION
**Pull request by erikcorry**
I think for something like this we want to use Kasper's new C++ code that
is used for RPC.  It's part of separating out the object heap and the
program heap so we can change the GC.

### CODE REVIEW

**Review by kasperl — APPROVED** [review]
Looks good to me.

File: src/primitive_core.cc [hunk 1]
```diff
@@ -445,6 +445,7 @@ PRIMITIVE(read_int_little_endian) {
 
 PRIMITIVE(args) {
   char** argv = process->args();
+#ifndef TOIT_FREERTOS
```
Comment (floitsch) [inline src/primitive_core.cc:448]: This feels like a completely unrelated change.
If we don't want to have compile-time arguments then we should remove them in a different PR.

File: src/snapshot.cc [hunk 2]
```diff
@@ -33,7 +33,7 @@ enum class SnapshotTypeTag {
   OBJECT_TAG = 0,
   IN_TABLE_TAG,
   BACK_REFERENCE_TAG,
-  PROGRAM_HEAP_REFERENCE_TAG,
+  UNUSED_TAG,
```
Comment (floitsch) [inline src/snapshot.cc:36]: I would prefer if it was removed, but since snapshots are disappearing soon anyway...

File: tools/toit.run.toit [hunk 3]
```diff
@@ -17,7 +17,6 @@
 // only: To make stack traces readable.
 
 import ar show *
-import debug.rpc show RPC_ECHO
 import .debug_message
```
Comment (floitsch) [inline tools/toit.run.toit:20]:
```suggestion
```

File: lib/debug/debug.toit [hunk 4]
```diff
@@ -5,10 +5,6 @@
 import .rpc
 import rpc
```
Comment (floitsch) [inline lib/debug/debug.toit:6]: Same for the 'echo.toit'
```suggestion
```

**Review by floitsch — APPROVED** [review]
LGTM.

File: src/objects.h [hunk 5]
```diff
@@ -460,6 +460,7 @@ class ByteArray : public HeapObject {
     *extra_bytes = raw_length;
   }
 
+#ifndef TOIT_FREERTOS
```
[the remark this answers is not in the document]
Comment (erikcorry) [reply src/objects.h:463]: Removed them instead.

File: src/primitive_core.cc (hunk 1 above)
Comment (erikcorry) [reply src/primitive_core.cc:448]: Change removed.

**Review by erikcorry — COMMENTED** [review]
All other changes implemented.
